ammo-fe/src/network/MockedSocket.ts

```diff
@@ -1,29 +1,29 @@
 // Arguments can be either a callback function (in the case of the toggleRecord for example), or a list of unknowned arguments
-type Arguments = unknown[] | ((recording: boolean) => void);
+type Arguments = unknown[];
 
 export default class MockedSocket {
     private cache: Record<string, (...args: unknown[]) => void> = {};
 
-    private isRecording = true;
+    public isSubbedToBullets = true;
 
     public connected = true;
 
     on = (event: string, listener: (...args: unknown[]) => void): void => {
         this.cache[event] = listener;
     };
 
-    emit = (event: string, args: Arguments): void => {
-        // When we toggle, we save the new value and invoke the callback given just like the server would do
-        if (event === 'toggleRecord') {
-            this.isRecording = !this.isRecording;
-            if (typeof args === 'function') {
-                args(this.isRecording);
-            }
-            return undefined;
+    emit = (event: string, args: Arguments, cbk: unknown): void => {
+        // Sub to bullets channel
+        if (event === 'bullets::sub') {
+            const { sub } = (args as unknown) as { sub: boolean };
```

Comment:
We already declare "args" as an array of unknown, why do you cast it as unknown ?

Reply:
I struggled a lot with typing. If you have a better syntax I'll take it 100%

Reply:
I might need to go and check the code then, not to sure

Reply:
I would add that the previous method already use it like so : 

```is
...args: unknown[]
```

So the extra type called Arguments might be useless

Reply:
Here is the full solution, I just tried it out. I'll send you the full file on Discord if you need a global view

```js

type SubEvent = 'bullets::sub';
type EmitEvent = 'bullets::emit';

type Events = SubEvent | EmitEvent;

type Sub = {
    sub: boolean;
};

type SubBullet = {
    bullet: Bullet;
};

// We made the mistake to think that .emit arguments are unknowned
type Arguments = Sub & SubBullet;

// This type is here to apply a kind of if/else on types
type Extract<E> = E extends SubEvent
    ? Sub // If the event is 'bullets::sub', then we return the Sub type
    : E extends EmitEvent // If it is 'bullets::emit', the SubBullet type
    ? SubBullet // And so on
    : unknown;

emit = (event: Events, args: Arguments, cbk: unknown): void => {
        // Sub to bullets channel
        if (event === 'bullets::sub') {
            const { sub }: Extract<typeof event> = args;
```